### ISSUE TITLE
Error r(111) in Stata MP 16.1 and SE 17.0, macOS Monterey

### ISSUE THREAD

**Issue by manuelmontesinos** [mcaceresb/stata-gtools#83]
I have recently installed `gtools` on my laptop by executing `ssc install gtools` and `gtools, upgrade`. However, when I try to reproduce some of the examples on the online documentation, I get an error:

```stata
. sysuse auto, clear
(1978 Automobile Data)

. gegen id = group(foreign), verbose bench(3)
Plugin setup; .001 seconds
Generated targets; .025 seconds
Parsed by variables; .002 seconds
r(111);

. glevelsof rep78, verbose bench(3)
Parsed by variables; .032 seconds
r(111);

```
I've got the same error in both Stata MP 16.1 and Stata SE 17.0. My OS is macOS Monterey, version 12.0.1. The version of `gtools` I have installed is 1.8.1 (27Aug2021).

**Comment by manuelmontesinos** [comment]
I don't know if this is useful, but I also tried to compile the plugin myself by following the instructions on the online documentation. When I run `do build/gtools_tests.do`, the `checks_gegen` test fails. This is the output I get:

```stata
-------------------------------------
Basic unit-tests 16:04:51 18 Nov 2021
-------------------------------------
    test(passed): checks_gcontract, oncollision(error)
    test(passed): checks_isid, oncollision(error)
    test(passed): checks_duplicates, oncollision(error)
    test(passed): checks_toplevelsof, oncollision(error)
    test(passed): checks_levelsof, oncollision(error)
    test(passed): checks_unique, oncollision(error)
    test(passed): checks_hashsort, oncollision(error)
    test(passed): checks_gregress, oncollision(error)
    test(passed): checks_greshape, oncollision(error)
    test(passed): checks_gstats, oncollision(error)
    test(passed): checks_gquantiles_by, oncollision(error)
    test(passed): checks_gquantiles_by, oncollision(error) wgt([fw = int1])
    test(passed): checks_gquantiles_by, oncollision(error) wgt([pw = int1])
    test(passed): checks_gquantiles_by, oncollision(error) wgt([aw = int1])
    test(passed): checks_gquantiles, oncollision(error)
    test(passed): checks_gquantiles, oncollision(error) wgt([fw = int1])
    test(passed): checks_gquantiles, oncollision(error) wgt([pw = int1])
    test(passed): checks_gquantiles, oncollision(error) wgt([aw = int1])
    test(failed): checks_gegen, oncollision(error)

WARNING: 16:48:38 18 Nov 2021
tests ran with non-0 exit status

tests ran but Stata gave error code r(198)

Start: 16:03:21 18 Nov 2021
End: 16:48:38 18 Nov 2021
```

**Comment by mcaceresb** [comment]
@manuelmontesinos Can you post the output of just `checks_gegen` if you are able?

Apologies for the compile issue. It seems to keep coming up; I'll make a new issue for this.

**Comment by manuelmontesinos** [comment]
This is what I get when I track the execution of the test: [test_gegen_output.txt](https://github.com/mcaceresb/stata-gtools/files/7568896/test_gegen_output.txt)

**Comment by mcaceresb** [comment]
Should be fixed in 6dc0fd6bbcc19e6f41f1095ffeaac39a51b7283f
